sdk/trace/batch_span_processor_test.go

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package trace_test
+package trace
```

Comment:
Instead of changing all the tests to be in the `trace` package, can we add a new test file that specifically tests the things we want to verify in the `trace` package?

Reply:
It wasn't clear to me why this was in the `trace_test` package in the first place.  Was this perhaps a vestige of prior import cycle conflicts that have been resolved, or is it done explicitly to avoid testing internals?

Reply:
IIRC it was done for the latter, to avoid testing internals. There were a few instances where we were originally testing external behavior from an internal test package that led to some errors (it was a while ago, but I can dig it up if we want to take a look at it).

That said, I'm not opposed to having tests that test internals, but I'm in favor of them being separate.